dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/tcp/LibertyLoggingHandler.java

```diff
@@ -0,0 +1,161 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.netty.internal.tcp;
+
+import java.net.SocketAddress;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.websphere.ras.annotation.Trivial;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+import io.netty.handler.logging.LoggingHandler;
+import io.netty.util.AttributeKey;
+
+@Trivial
+public class LibertyLoggingHandler extends LoggingHandler{
+
+	protected static final TraceComponent tc = Tr.register(LibertyLoggingHandler.class, new String[]{TCPMessageConstants.TCP_TRACE_NAME,TCPMessageConstants.NETTY_TRACE_NAME},
+			TCPMessageConstants.TCP_BUNDLE, LibertyLoggingHandler.class.getName());
+	
+	protected final static AttributeKey<Integer> BYTES_READ_KEY = AttributeKey.valueOf("BYTES_READ");
+
+	public LibertyLoggingHandler(){
+		super(LibertyLoggingHandler.class);
+		//TODO Check mapping between logger level in Netty and TraceComponent
+	}
+	
+	@Override
+	public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+			Tr.event(ctx.channel(), tc, "handler added to" + ctx + ", channel " + ctx.channel());
+		}
+		ctx.channel().attr(BYTES_READ_KEY).getAndSet(0);
+		super.handlerAdded(ctx);
+	}
+
+	@Override
+	public void channelRegistered(ChannelHandlerContext ctx) throws Exception {
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+			Tr.event(ctx.channel(), tc, "registered " + ctx.channel() + ", key is " + null); //TODO Can't get selection key for Netty, check if this is actually necessary
```

Comment:
If key is not obtainable at this point, it is likely better to not include it. Having it say 'key is null' might be confusing from an L2 perspective.

Reply:
I agree, I'll make the change to remove this